ci/ci

```diff
@@ -80,23 +151,37 @@ def run_cmd(cmd, return_output=False):
         log_error(e)
 
 
-def run_docker_cmd(cmd):
+def run_docker_cmd(run_cmd, only_run_on='all'):
     '''
-    Runs a docker command and on failure redirects user to the container terminal
+    Runs a docker command and on failure redirects user to
+    the container terminal
     '''
-    try:
-        docker_cmd = shlex.split(cmd)
-        a = subprocess.Popen(docker_cmd)
-        a.communicate()
-        if a.returncode != 0:
-            log_error(
-                "docker cmd returned with non zero exit code, redirecting you to container terminal")
-            docker_cmd = shlex.split(
-                "docker-compose exec pbs-build bash -c \'cd /pbssrc && /bin/bash\'")
-            subprocess.run(docker_cmd)
-    except Exception as e:
-        log_error("Failed\n:")
-        log_error(e)
+    services = get_services_list()
+    services.sort(reverse=True)  # we want server cmds to run first
+    for service in services:
+        cmd = "docker-compose -f docker-compose.json exec " + \
+            service + " bash -c \'" + run_cmd + "\'"
+        if only_run_on != 'all' and service.find(only_run_on) == -1:
+            log_info('Skipping on ' + service +
+                     ' as command only to be run on ' + only_run_on)
+            continue
+        try:
+            log_info(cmd)
+            docker_cmd = shlex.split(cmd)
+            a = subprocess.Popen(docker_cmd)
+            a.communicate()
+            if a.returncode != 0:
+                log_error(
+                    "docker cmd returned with non zero exit code, \
+                    redirecting you to container terminal")
+                docker_cmd = shlex.split(
+                    "docker-compose -f docker-compose.json exec " +
+                    service + " bash -c \'cd /pbssrc && /bin/bash\'")
+                subprocess.run(docker_cmd)
+                os._exit(1)
```

Comment:
why exit here? is this child? no right?

Reply:
We do not want to continue with execution if any command has failed. If we don't use os._exit(1) here ; when user exits from container the process continues.... for example configuration failed , user was redirected to container terminal but when the user exits the container terminal it will continue with make command.